tools/Google.Cloud.Tools.ReleaseManager/NonSourceGenerator.cs

```diff
@@ -0,0 +1,689 @@
+        }
+
+        var linesBefore = templateLines.Take(tableIndex).ToList();
+        var linesAfter = templateLines.Skip(tableIndex + 1).ToList();
+
+        var table = new List<string>
+        {
+            "| Package | Latest version | Description |",
+            "|---------|----------------|-------------|"
+        };
+
+        var ambiguousDescriptions = _catalog.Apis
+            .Select(api => api.EffectiveListingDescription)
+            .GroupBy(description => description)
+            .Where(g => g.Count() > 1)
+            .Select(g => g.Key)
+            .ToList();
+        foreach (var api in _catalog.Apis)
+        {
+            // TODO: What about 2.0.0-beta00 etc? We'd need to know what version to link to.
+            // We can cross that bridge when we come to it.
+            if (api.Version == "1.0.0-beta00" || api.Version == "1.0.0-alpha00" || api.ReleaseLevelOverride == "unreleased")
+            {
+                continue;
+            }
+            string url = ApiMetadata.IsCloudPackage(api.Id)
+                ? $"https://cloud.google.com/dotnet/docs/reference/{api.Id}/latest"
+                : $"https://googleapis.dev/dotnet/{api.Id}/latest";
+            string referenceDocsLink = $"[{api.Id}]({url})";
+
+            string packageLinkAndBadge = $"[![NuGet](https://img.shields.io/nuget/v/{api.Id})](https://nuget.org/packages/{api.Id})";
+
+            // Disambiguate direct API packages for the same product.
+            string description = api.EffectiveListingDescription;
+            if (ambiguousDescriptions.Contains(description) && api.ApiVersion is string apiVersion)
+            {
+                description += $" ({apiVersion} API)";
+            }
+
+            // TODO: It gets awkward when there are multiple description for the same product (e.g. Spanner).
+            // This is basically caused by conflating "link to product docs" with "package description", but
+            // in most cases that simplifies things. Never mind.
+            string productLink = api.ProductUrl is null
+                ? api.ListingDescription ?? api.ProductName ?? api.Description.TrimEnd('.') // No URL
+                : $"[{description}]({api.ProductUrl})";
+            table.Add($"| {referenceDocsLink} | {packageLinkAndBadge} | {productLink} |");
+        }
+
+        var newContent = linesBefore.Concat(table).Concat(linesAfter);
+        var outputPath = Path.Combine(_outputDirectory, "README.md");
+        File.WriteAllLines(outputPath, newContent);
+    }
+
+    private void RewriteRenovate()
+    {
+        string templatePath = Path.Combine(_generatorInputDirectory, "renovate-template.json");
+        string json = File.ReadAllText(templatePath);
+        JObject jobj = JObject.Parse(json);
+        JArray ignorePaths = (JArray) jobj["ignorePaths"];
+        foreach (var api in _catalog.Apis)
+        {
+            ignorePaths.Add($"apis/{api.Id}/{api.Id}/**");
```

Comment:
Duh, I left a comment somewhere about why renovate was in input.

Maybe there's a way to configure renovate so that we don't have to be explicit about each folder we want to skip? Maybe we can just add the ones we want renovate to look at.